tests/conftest.py

```diff
@@ -506,3 +506,40 @@ def teardown_yum_content():
         assert teardown.values()[0]["rc"] == 0
 
     request.addfinalizer(teardown_yum_content)
+
+
+@pytest.fixture(scope="function")
+def setup_custom_package(request, ansible_module):
+    """ Setup/Teardown cusom yum repo/package for non-rh-packages check."""
+    ansible_module.yum_repository(
+        name="custom_repo",
+        description="custom repo",
+        file="custom_repo",
+        baseurl=f"{FAKE_YUM0_REPO}",
+        enabled="yes",
+        gpgcheck="no",
+    )
+    setup = ansible_module.file(path="/etc/yum.repos.d/custom_repo.repo", state="present")
+    assert setup.values()[0]["changed"] == 0
+
+    pkgs_locked = ansible_module.command(Packages.is_locked()).values()[0]["rc"]
+    if pkgs_locked == 0:
+        ansible_module.command(Packages.unlock())
```

Comment:
@Gauravtalreja1 basically you are running this test when FM Packages lock is disabled. I don't think this is right way to test this. You should enable package lock after installing the `walrus` package.

Reply:
@jameerpathan111 Updated. request to re-review.